src/algos/fl_static.py

```diff
@@ -42,20 +43,26 @@ def run_protocol(self) -> None:
         epochs_per_round = self.config.get("epochs_per_round", 1)
         for it in range(start_round, total_rounds):
             # Train locally and send the representation to the server
-            stats["train_loss"], stats["train_acc"] = self.local_train(
-                epochs_per_round
-            )
+            is_working = self.get_and_set_working(it)
+            if is_working:
+                stats["train_loss"], stats["train_acc"] = self.local_train(
```

Comment:
I think this skipping the training part should go to `local_train` otherwise every algorithm will have to write the same code.

Reply:
Migrated this into the base classes.